### PR TITLE
[WIP] Improvements of GitHub script

### PR DESCRIPTION
**Pull request by krzyurb**
**Fixes and changes:**

* `pr` is now an alias of `pull-request`
* `github pull-request bob-the-bob` without specified pull request id responds with all open pull requests for repository
* `github hygiene` command can be used for specific slack user or for the whole repository, e.g. 
`github hygiene bob-the-bob` - for whole repo and `github hygiene bob-the-bob @john` for user

I also moved code from `github/index.js` into separated `GithubController` class and split `github.spec.js` into separated files to make tests more readable. I also check if slack user github field exists.

### CODE REVIEW

File: src/github/index.js [hunk 1]
```diff
@@ -1,66 +1,16 @@
...
-      return res.send('Couldn\'t get data from github');
-    }
-  };
-
-  const githubHelp = (res) => {
-    res.send(new MessageRenderer('github/help').render()); // TODO
-  };
-
   robot.respond(/github .*/i, async (res) => {
+    const controller = new GithubController(robot);
```
Comment (Davlis) [inline src/github/index.js]: I would highly appreciate if you could pass services and Logger to github controller nstead of `robot` object

```javascript
// index.js
const { PullRequestService, ProfileService } = robot.brain
const { logger } = robot;
```

```javascript
// github.controller.js
constructor(ProfileService, PullRequestService, Logger) {
  this.ProfileService = ProfileService;
  this.PullRequestService = PullRequestService;
  this.Logger = Logger;
}
```

Why?
- Easier to mock
- Probably we will user our logger in future to handle errors from RespondMultiplexer and it's responses, so we won't use `robot.logger`
- We're decoupling `hubot` from controller

File: src/github/HygieneCheck.service.js [hunk 2]
```diff
@@ -10,12 +10,16 @@ class HygieneCheckService {
...
   }
 
   getOwnerRepositoryName(query) {
     const match = query.match(/(.*)\/(.*)/);
-    return (match) ? `${match[1]}/${match[2]}` : `${process.env.HUBOT_GITHUB_ORG}/${query}`;
+    return (match) ? `+${match[1]}/${match[2]}` : `${process.env.HUBOT_GITHUB_ORG}/${query}`;
+  }
+
+  getPullRequestsScope(githubUser) {
+    return (githubUser) ? `+involves:${githubUser}` : '';
```
Comment (Davlis) [inline src/github/HygieneCheck.service.js]: I believe that we don't need brackets here

File: src/github/HygieneCheck.service.js [hunk 3]
```diff
@@ -76,8 +84,9 @@ class HygieneCheckService {
     return config.hygiene;
   }
 
-  async checkHygiene(repositoryName, githubUrl) {
-    const githubUser = this.extractUserName(githubUrl);
+  async checkHygiene(repositoryName, githubUrl = null) {
+    const githubUser = (githubUrl) ? this.extractUserName(githubUrl) : null;
```
Comment (Davlis) [inline src/github/HygieneCheck.service.js]: Brackets! 😅

File: src/github/HygieneCheck.service.js (hunk 2 above)
Comment (krzyurb) [reply src/github/HygieneCheck.service.js]: Dunno why, but for me it looks nicer 😄But they are not necessary

**Comment by krzyurb** [comment]
@Davlis  changes done. I believe those codeclimate issues will disappear when we will change `RespondMultiplexer`

**Pull request merged**